### ISSUE TITLE
iPad 15.1 Users kicked out of meeting when turning on Camera in meeting

### ISSUE THREAD

**Issue by BC-Mikey** [aws/amazon-chime-sdk-js#1762]
### What happened and what did you expect to happen?

iPad users are able to allow their cameras and see them during the Camera Preview. However, after they enter the meeting they 

This was only reproducible on version 15.1. Versions 15.0 and lower worked fine. 

### Have you reviewed our existing documentation?

- [X] GitHub [issues](https://github.com/aws/amazon-chime-sdk-js/issues)
- [X] FAQs [page](https://aws.github.io/amazon-chime-sdk-js/modules/faqs.html)
- [X] README [page](https://github.com/aws/amazon-chime-sdk-js#amazon-chime-sdk-for-javascript)
- [X] Feature-specific [guides](https://github.com/aws/amazon-chime-sdk-js#guides)

### Reproduction steps

1) Log into a meeting on an iPad v15.1
2) You can have other users join the meeting & share their cameras, but this shouldn't be necessary. The issue was reproduced in a meeting alone.
3) Attempt to turn on your camera from your iPad
4) Instead of sharing your camera, you will be kicked out of the meeting.

### Amazon Chime SDK for JavaScript version

2.21.0

### What browsers are you seeing the problem on?

Safari iOS

### Browser version

15.0

### Meeting and Attendee ID Information.

Meeting ID: bc-chime-test2

### Browser console logs

[ipad-kick-out-bug-log.txt](https://github.com/aws/amazon-chime-sdk-js/files/7500134/ipad-kick-out-bug-log.txt)

**Comment by ltrung** [comment]
This is a duplicate of #1710 and is currently tracked in the known browser bug issue in #1059. Can you try to pull this branch to see if it works for you https://github.com/aws/amazon-chime-sdk-js/tree/ios-15.1-fix ?

**Comment by BC-Mikey** [comment]
@ltrung We tested the ios-15.1-fix branch and were still able to reproduce the issue. I'm attaching our logs.
[ipad-test-log.txt](https://github.com/aws/amazon-chime-sdk-js/files/7514961/ipad-test-log.txt)

**Comment by ltrung** [comment]
In the log I saw this error message which likely the cause of the crash. Content Share does not work in iOS since they do not support the webRTC screen capture API `getDisplayMedia`
```
[Error] 2021-11-10T17:34:16.920Z [ERROR] SDK - Could not start content share: TypeError: navigator.mediaDevices.getDisplayMedia is not a function. (In 'navigator.mediaDevices.getDisplayMedia(e)', 'navigator.mediaDevices.getDisplayMedia' is undefined)
	log (04f0-149-20-194-53.ngrok.io:2:176799)
	error (04f0-149-20-194-53.ngrok.io:2:176418)
	error (04f0-149-20-194-53.ngrok.io:2:180446)
	(anonymous function) (04f0-149-20-194-53.ngrok.io:2:1006152)
	generatorResume
	a (04f0-149-20-194-53.ngrok.io:2:942299)
	promiseReactionJob
```

**Comment by BC-Mikey** [comment]
I confirmed that they are turning on the camera. We just tried another test in the browser demo application, making sure that he clicked the Camera icon 
[iPad15_1_logs_2.txt](https://github.com/aws/amazon-chime-sdk-js/files/7515286/iPad15_1_logs_2.txt)

**Comment by BC-Mikey** [comment]
As far as our application goes, we detect mobile devices and disable the Screen Share button.

**Comment by rachinkapoor** [comment]
We are also testing the fix. Seems to be working for us if and only if `Request Desktop Website` option is disabled.
By default all iPads have this option turned on.

**Comment by ltrung** [comment]
@rachinkapoor Thanks. So this is indeed the same issue as #1710 as I mentioned above. We disabled H.264 for now for iOS device based on user agent but i guess when iPad has `Request Desktop Website` on, the user agent will be Mac Safari and not iOS thus, we did not do the workaround.

**Comment by rachinkapoor** [comment]
As this is impacting us big time, I need your advice.

**Comment by ltrung** [comment]
As a workaround, can you retry the code here https://github.com/aws/amazon-chime-sdk-js/commit/e15565d4cff205ce69c44bdccc7b2f2a5c48a991#diff-12999dcd6545b6debd7a9cb6c9a75abb4982bf367947a9a2e97fbc87c1aa1e10R41 and update this to include Safari also https://github.com/aws/amazon-chime-sdk-js/commit/e15565d4cff205ce69c44bdccc7b2f2a5c48a991#diff-878aade3f1ff249700b52b8c36f7715a1fd27b4f73344d55072e08aae78b8608R277

**Comment by ltrung** [comment]
Updated the code in the branch https://github.com/aws/amazon-chime-sdk-js/tree/ios-15.1-fix to include Safari and iPad 15.1.
Can you give it a try?

**Comment by dchen-aws** [comment]
See response here also https://github.com/aws/amazon-chime-sdk-js/issues/1710#issuecomment-971039472

**Comment by karthi-manoharan** [comment]
Hi Team, 

We are waiting for the issue to be fixed. It's creating problems with the iPad and now iPad users cant able to use the application for the past 9 days. We are expecting a quick fix asap.

**Comment by BissacM** [comment]
Still have not understood if this issue was fixed or not.

**Comment by dchen-aws** [comment]
@karthi-manoharan @BissacM this [branch](https://github.com/aws/amazon-chime-sdk-js/tree/ios-15.1-fix) contains a fix for Chrome and Safari on iPadOS 15.1.

**Comment by pgutierrez-bcov** [comment]
Tried the latest from master as it seems to have been merged and we were still able to reproduce this on iPadOS15.1. Works fine on iOS15.1

**Comment by ltrung** [comment]
It worked fine for me on iPad iOS 15.1.0. Could you attach a console log?

**Comment by dchen-aws** [comment]
> Tested the latest from master as it seems that the fix has been merged and we were still able to reproduce this on iPadOS15.1. Works fine on iOS15.1

and which browser are you using?

**Comment by ezramorse** [comment]
Please bump npm to 2.21.2. Thanks

**Comment by karthi-manoharan** [comment]
Hi @ltrung

**Comment by ltrung** [comment]
Double check and verified that I could turn it on in my iPad Pro.
@karthi-manoharan  Could you print out the navigator.userAgent in that iPad and post it here and maybe a console log before it crashes? The core issue is that iOS 15.1 has a bug with H264 video encoding so we need to disable it for certain devices based on user agent so my guess is that your device has a different user agent than the one I tested.

**Comment by karthi-manoharan** [comment]
@ltrung 
As per iPad, I can able to debug only the Safari browser. I cant able to debug and get the console log for the brave browser. Can you please help me to find out

**Comment by ltrung** [comment]
@karthi-manoharan Please refer to my previous post but we do not currently support Brave information.

**Comment by karthi-manoharan** [comment]
Thanks @ltrung

**Comment by ltrung** [comment]
The workaround fix for iOS 15.1 has been released in the latest NPM version so closing this issue.
